Mathlib/Algebra/Module/LinearMap.lean

```diff
@@ -576,14 +576,22 @@ theorem comp_assoc
 
 variable {f g} {f' : M₂ →ₛₗ[σ₂₃] M₃} {g' : M₁ →ₛₗ[σ₁₂] M₂}
 
+/-- The linear map version of `Function.Surjective.injective_comp_right` -/
+lemma inj_comp_right_of_surj (hg : Surjective g) : Injective fun f : M₂ →ₛₗ[σ₂₃] M₃ ↦ f.comp g :=
+  fun _ _ h ↦ ext <| hg.forall.2 (ext_iff.1 h)
```

Comment:
I added this docstring, but the name is still yours. I don't think using `_inj` and `_surj` like this is particularly standard, so perhaps:
```suggestion
/-- The linear map version of `Function.Surjective.injective_comp_right` -/
lemma injective_comp_right_of_surjective (hg : Surjective g) : Injective fun f : M₂ →ₛₗ[σ₂₃] M₃ ↦ f.comp g :=
  fun _ _ h ↦ ext <| hg.forall.2 (ext_iff.1 h)
```
or
```suggestion
/-- The linear map version of `Function.Surjective.injective_comp_right` -/
lemma _root_.Function.Surjective.injective_linearMapComp_right (hg : Surjective g) : Injective fun f : M₂ →ₛₗ[σ₂₃] M₃ ↦ f.comp g :=
  fun _ _ h ↦ ext <| hg.forall.2 (ext_iff.1 h)
```
or something in between...

Reply:
Yeah I was on the fence about the naming. I've renamed with your (second) suggestion.

Reply:
Thanks!